libs/util/http.client.js

```diff
@@ -211,6 +211,9 @@ function io(url, options) {
         resp.responseText = body;
 
         if (err) {
+            if (err.message === 'XMLHttpRequest timeout') {
+                err = new Error('XMLHttpRequest timeout: failed to load ' + resp.url + ' in ' + options.timeout + 'ms');
+            }
```

Comment:
I was able to solve my problem with `statsCollector`.
I will create a new PR to always set `err.timeout`.